File: app/src/main/java/com/example/skillswapexchange/network/ApiService.java
package com.example.skillswapexchange.network;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface ApiService {
    @FormUrlEncoded
    @POST("register.php")
    Call<ApiResponse> register(
        @Field("fullname") String fullname,
        @Field("email") String email,
        @Field("password") String password,
        @Field("mobile") String mobile
    );

    @FormUrlEncoded
    @POST("login.php")
    Call<ApiResponse> login(
        @Field("email") String email,
        @Field("password") String password
    );
}

class ApiResponse {
    public String status;
    public String message;
}

File: app/src/main/java/com/skillexchange/app/api/ApiService.kt
package com.skillexchange.app.api

import retrofit2.Response
import retrofit2.http.*

interface ApiService {
    @FormUrlEncoded
    @POST("api/auth/login")
    suspend fun login(
        @Field("username") username: String,
        @Field("password") password: String
    ): Response<ApiResponse<UserDto>>

    @POST("api/auth/register")
    suspend fun register(
        @Body user: UserDto
    ): Response<ApiResponse<UserDto>>

    @GET("api/skills")
    suspend fun getSkills(
        @Query("category") category: String? = null,
        @Query("type") type: String? = null,
        @Query("search") search: String? = null,
        @Query("userId") userId: String? = null
    ): Response<List<SkillDto>>

    @POST("api/skills")
    suspend fun createSkill(
        @Body body: CreateSkillBody
    ): Response<SkillDto>

    @DELETE("api/skills/{id}")
    suspend fun deleteSkill(
        @Path("id") skillId: String
    ): Response<Void>

    @GET("api/requests")
    suspend fun getRequests(
        @Query("userId") userId: String
    ): Response<Map<String, List<RequestDto>>>

    @POST("api/requests")
    suspend fun createRequest(
        @Body body: CreateRequestBody
    ): Response<RequestDto>

    @PUT("api/requests/{id}")
    suspend fun updateRequestStatus(
        @Path("id") requestId: String,
        @Body body: Map<String, String>
    ): Response<RequestDto>

    @GET("api/messages")
    suspend fun getMessages(
        @Query("userId") userId: String,
        @Query("peerId") peerId: String
    ): Response<List<MessageDto>>

    @POST("api/messages")
    suspend fun sendMessage(
        @Body message: MessageDto
    ): Response<MessageDto>

    @POST("api/feedback")
    suspend fun submitFeedback(
        @Body feedback: FeedbackBody
    ): Response<ApiResponse<String>>

    @GET("api/status")
    suspend fun getStatus(): Response<Map<String, Any>>
}
